src/server/replica.cc

```diff
@@ -142,18 +142,29 @@ bool Replica::Start(ConnectionContext* cntx) {
   ProactorBase* mythread = ProactorBase::me();
   CHECK(mythread);
 
+  auto should_continue = [this, &cntx](const error_code& ec, const char* msg) {
+    if (state_mask_ & R_CANCELLED) {
+      (*cntx)->SendError("replication cancelled");
+      return false;
+    }
```

Comment:
The flag is a valid solution, but we already have the context for such management - you can check with `cntx_.IsCancelled()` whether its cancelled.

To catch the cancellation mid flight, `Switch` should be used with the context below, not `Reset`. The first one checks for pending cancellation and returns it

Reply:
thanks! changed it :)